pp-2023/src/app/kmeans/calculateButton.js

```diff
@@ -20,62 +21,69 @@ export function HandleCalculateButtonClick() {
     Die Funktion handleClick steuert als Controller die Anwendungslogik, welche Daten verwendet werden und wo diese verarbeitet werden.
      */
     const handleClick = () => {
-        const kPoints = validateKPoints(numberOfClusters);
-        //const inputDataSrc = checkInputSource(); TODO
-        //const  localCalculation = checkLocalOrServer(); TODO
-        const localCalculation = false; // nur zum Testen
-        const inputDataSrc = 'file'; // nur zum Testen
-        const dataArrayForWorking = inputDataArray;
-        chartDeletion = 1; //gibt an, dass das alte Chart von der ScatterChart funktion gelöscht werden muss
-
-        if (inputDataSrc === "file") {
-            /*
-            Verarbeitung eines Files und Export an die KMeans-Api, anschließend wird das Ergebnis visualisiert.
-             */
-            if (!localCalculation) {
-                const resultPost = apiPostRequest(kPoints, false);
-                console.log(resultPost); // TODO handling muss noch gemacht werden
-return;
-                const resultGetStateOfTask = apiGetStateOfTask();
-                resultGetStateOfTask.then(result => {
-
-                    if (result === 200) {
-                        const resultKMeans = apiGetResult();
-                        resultKMeans.then(resultKMeans => {
-                            console.log(resultKMeans.result);
-                            //erzeugt das 2d Chart mit hilfe der Berechneten Daten des kMeans Algorithmus
-                            ScatterChart(numberOfClusters,chartDeletion,result);
-                        });
-                    }
-                });
+        // hier wird auf die Daten der excel gewartet
+        calculateexcel((exceldata) => {
+            // sobald die Daten eingelesen wurden gehts weiter
+
+            const kPoints = validateKPoints(numberOfClusters);
+            //const inputDataSrc = checkInputSource(); TODO
+            //const  localCalculation = checkLocalOrServer(); TODO
+            const localCalculation = false; // nur zum Testen
+            const inputDataSrc = 'file'; // nur zum Testen
```

Comment:
Bitte die Funktion checkInputSrc nutzen und entsprechend file zurückgeben wenn eins gefunden wurde !

Reply:
Ist doch genau das was ReturnExcel() macht? Willst du das ich die Funktion in checkInputSrc umbenenne oder was?

Reply:
Ja das muss in der Funktion passieren . Die Funktion soll steuern ob ein File oder eine Manuelle Eingabe da ist.